### ISSUE TITLE
Ellipses operator in optional arguments

### ISSUE THREAD

**Issue by qguv** [docopt/docopt#168]
I cannot seem to get the `...` operator to parse correctly within optional arguments. I am [migrating](https://github.com/qguv/rpcalc/commit/97b4893b425be12b0b67ec65dc9bcf742223b84d) my project from argparse to docopt. The docstring is as follows:

<pre>
rpcalc, a reverse polish notation calculator

Usage:
  rpcalc [-s N] [-i X...]
  rpcalc [-e] -i X...
  rpcalc (-h | --help)
  rpcalc --version

Options:
  -h --help     Show this screen.
  -s N          Limit length of stack to N elements.
  -i X...       Push following elements to stack.
  -e            Limit stack length to amount of elements given with -i.
  --version     Display version.
</pre>

I get the following cryptic error message from docopt:

<pre>
$ python3 rpcalc.py -i 1 2 3
Usage:
  rpcalc [-s N] [-i X...]
  rpcalc [-e] -i X...
  rpcalc (-h | --help)
  rpcalc --version
</pre>

It looks like `[-i X...]` in the first _Usage_ line is what's causing the issue.
**TL;DR** Can docopt handle the ellipses operator within `[` `]`?

**Comment by zcrisler** [comment]
Well, argparse allows options to specify multiple arguments via `nargs='*'` and `nargs='+'`. Can you elaborate on why you think it is a misfeature? I think it is a very common idiom.

**Comment by keleshev** [comment]
Because in invocation like `prog --foo bar baz qux` there's no way for a person to tell if it means `prog --foo=bar baz qux` or `prog --foo=bar,baz qux` or `prog --foo=bar,baz,qux`.

**Comment by fsaintjacques** [comment]
:+1: with @keleshev, the `nargs='*'` and `nargs='+'` are dangerous feature that leads to ambiguity.

**Comment by alvapan** [comment]
I have a use case for an option with more than one arguments:

```python
 """Find duplicated files with same hashed value.
Usage: finddup.py [options] [<path>...] [--skip <skip>...]
<path> can be file or directory name.
Options:
  -h  --help           show this help message and exit
  --version            show version info
  -e EXT --ext=EXT
  -s SIZE --size=SIZE  search for dup. files with some size
  --min=MINSIZE        search for dup. files greater than or equal to some size
  --max=MAXSIZE        search for dup. files less than or equal to some size
  -S --skip <skip>...  skip some paths, <skip> can include wildcards
"""
```

By appending `--skip` option after the `<path>...` positional arguments, there won't be ambiguities if we give `--skip` more than one `<skip>` argument.
Since `<skip>` is assumed to be able to accept wildcards, such as `*~.txt`,
use multiple `--skip` options will not work.

**Comment by fsaintjacques** [comment]
I would suggest to reverse `<path>` and `--skip` like this
```
Usage: finddup.py [options]  [--skip <skip>...] [<path>...]
```
not only this is more standard (options before arguments), it follows what @keleshev said where you should really call your program with `finddup.py --skip=file.a --skip=file.b file.c file.d

**Comment by casperdcl** [comment]
I have implemented a `docopt` to `argparse` module called `argopt` (https://pypi.python.org/pypi/argopt#usage or https://github.com/casperdcl/argopt#usage)... It's pretty useful for extending syntax of `docopt` as well (just define most of the simple arguments in a docstring, then add advanced arguments such as muti-arg optionals manually).
